internal/services/rdb/instance.go

```diff
@@ -538,9 +559,25 @@ func ResourceRdbInstanceRead(ctx context.Context, d *schema.ResourceData, m inte
 	_ = d.Set("logs_policy", flattenInstanceLogsPolicy(res.LogsPolicy))
 
 	// set endpoints
+	var privateIP []map[string]interface{}
 	if pnI, pnExist := flattenPrivateNetwork(res.Endpoints); pnExist {
 		_ = d.Set("private_network", pnI)
+
+		if res.Endpoints[0].PrivateNetwork.ProvisioningMode == rdb.EndpointPrivateNetworkDetailsProvisioningModeIpam {
```

Comment:
Can the first endpoint be something else than a private network ? `res.Endpoints[0].PrivateNetwork` may be nil